### PR TITLE
checkers: teach boolExprSimplify a few new tricks

### PR DESCRIPTION
**Pull request by Quasilyte**
New patterns:

  `x <= y-1` => `x < y`
  `x+1 <= y` => `x < y`
  `x > y-1` => `x >= y`
  `x+1 > y` => `x >= y`
  `x >= y+1` => `x > y`
  `x-1 >= y` => `x > y`
  `x < y+1` => `x <= y`
  `x-1 < y` => `x <= y`

Updates #616

Signed-off-by: Iskander Sharipov <quasilyte@gmail.com>

### CODE REVIEW

File: checkers/boolExprSimplify_checker.go [hunk 1]
```diff
@@ -171,6 +172,61 @@ func (c *boolExprSimplifyChecker) combineChecks(cur *astutil.Cursor) bool {
...
+			cmp.Op = replacement
+			cur.Replace(cmp)
+			return true
+		case matchOneWay(rhsOp, rhs, lhs):
+			cmp.Y = rhs.X
+			cmp.Op = replacement
+			cur.Replace(cmp)
+			return true
+		}
+		return false
```
Comment (cristaloleg) [inline checkers/boolExprSimplify_checker.go]: Can be pulled into the switch 😉

File: checkers/boolExprSimplify_checker.go [hunk 2]
```diff
@@ -171,6 +172,61 @@ func (c *boolExprSimplifyChecker) combineChecks(cur *astutil.Cursor) bool {
...
+		// `x-1 < y` => `x <= y`
+		return replace(token.SUB, token.ADD, token.LEQ)
+
+	case token.LEQ:
+		// `x <= y-1` => `x < y`
+		// `x+1 <= y` => `x < y`
+		return replace(token.ADD, token.SUB, token.LSS)
+	}
+
+	return false
```
Comment (cristaloleg) [inline checkers/boolExprSimplify_checker.go]: Also into the switch

File: checkers/testdata/boolExprSimplify/negative_tests.go [hunk 3]
```diff
@@ -47,3 +47,16 @@ func floatCompare() {
 	// Can't be simplified to `f1 != f2`.
 	_ = !(f1 == f2)
 }
+
+func balancedIncDec(x, y, z int) {
+	// This is usually done on purpose.
```
Comment (cristaloleg) [inline checkers/testdata/boolExprSimplify/negative_tests.go]: What purpose? 🤔 Just curious.

Comment (Quasilyte) [reply checkers/testdata/boolExprSimplify/negative_tests.go]: Making expressions in a row (multi-statement) look more beautiful.
This is something like "balancing" them.
Same reason why someone writes +0 in expressions:
```go
x+0 < a+0
x+1 < b+1
x+2 < b+2
```

Comment (cristaloleg) [reply checkers/testdata/boolExprSimplify/negative_tests.go]: Ah, context.. thanks for the clarification.

**Pull request merged**